.github/workflows/linting.yml

```diff
@@ -71,3 +71,94 @@
         uses: actions/checkout@v4
       - name: Check DAOS ftest tags.
         run: \[ ! -x src/tests/ftest/tags.py \] || ./src/tests/ftest/tags.py lint
+
+  flake8-lint:
+    runs-on: ubuntu-22.04
+    name: Flake8 check
+    steps:
+      - name: Check out source repository
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+      - name: Set up Python environment
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3'
+      - name: Add parser
+        run: echo -n "::add-matcher::ci/daos-flake-matcher.json"
+      - name: Add whitespace parser
+        run: echo -n "::add-matcher::ci/daos-flakew-matcher.json"
+      - name: Add error parser
+        run: echo -n "::add-matcher::ci/daos-flakee-matcher.json"
+      - name: flake8 Lint
+        uses: py-actions/flake8@v2
+        with:
+          # W503 and W504 are related as they conflict.  W503 is the preferred style and all code
+          # should be using it now.
+          ignore: 'W503'
+          exclude: 'src/control/vendor,src/client/pydaos/raw'
+          max-line-length: '100'
+      - name: flake8 Lint on SCons files.
+        uses: py-actions/flake8@v2
+        with:
+          ignore: 'F821,W503,F841'
+          max-line-length: '100'
+          args: '--filename */SConscript, SConstruct'
+
+  Doxygen:
+    name: Doxygen
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+      - name: Install doxygen
+        run: sudo apt-get install doxygen
+      - name: Add parser
+        run: echo -n "::add-matcher::ci/daos-doxygen-matcher.json"
+      - name: Run check
+        run: doxygen Doxyfile
+      - name: 'Upload Artifact'
+        uses: actions/upload-artifact@v4
+        with:
+          name: API Documentation
+          path: docs/doxygen/html/
+          retention-days: 1
+
+  pylint:
+    name: Pylint check
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+      - uses: actions/setup-python@v5
+        with:
+          python-version: '3.11'
+      - name: Install core python packages
+        run: python3 -m pip install --requirement requirements.txt
+      - name: Install extra python packages
+        run: python3 -m pip install --requirement utils/cq/requirements.txt
+      - name: Install enchant
```

Comment:
I think as an experiment we should dismiss these ones as "Used in tests" as this code only runs on github hosted runners.

Reply:
That will only dismiss the GitHub alert raised.  It will do nothing about the Scorecard report and IPAS' general requirement that **every single** dependency be pinned to a git hash.

Reply:
But then shouldn't we dismiss all warnings for all GHA that run on GH runners? Then what's the point  in the OSSF requirements?
Not questioning your suggestion, but rather the feasibility of all of this as @brianjmurrell pointed out.

This dismissal description doesn't make sense to me
> These alerts are not in production code

None of our GHA are used in production code. I'm not sure what a "production GHA" would even be

Reply:
I was building on what I thought you'd said about these being warnings, not errors.

> That will only dismiss the GitHub alert raised. It will do nothing about the Scorecard report and IPAS' general requirement that **every single** dependency be pinned to a git hash.

I was building on what I thought you'd said about these being warnings, not errors.

Let's leave this for now then pending further investigation.